conans/test/integration/cache/download_cache_test.py

```diff
@@ -199,8 +199,8 @@ def source(self):
 
             assert 2 == len(os.listdir(os.path.join(tmp_folder, "s")))
             content = json.loads(load(os.path.join(tmp_folder, "s", sha256 + ".json")))
-            assert "http://localhost:5000/myfile.txt" in content["unknown"]
-            assert len(content["unknown"]) == 1
+            assert "http://localhost:5000/myfile.txt" in content["references"]["unknown"]
```

Comment:
Maybe we should add a test that using name and version in the command line or in the recipe changes this unknown to an actual reference.

I am also changing my mind, to upload it anyway even if unknown. There is the URL and the sha256, not really relevant if using only from a consumer without ref?

Reply:
For source it does not change, the conanfile's `_conan_node` will still be empty, so even if `name` and `version` are set, calling `ref` will still throw an `AttributeError` exception and so we'll still identify it as "unknown" :/